lisa/nic.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import ipaddress
+import itertools
+import re
+from typing import Dict, List
+
+from assertpy import assert_that
+
+from lisa import Node
+
+
+class NicInfo:
+
+    # Class for info about an single upper/lower nic pair.
+    # devices using SRIOV on azure typically have an upper synthetic device
+    # paired with a lower SRIOV Virtual Function (VF) device that
+    # enables the passthrough to the physical NIC.
+
+    def __init__(
+        self,
+        upper: str,
+        lower: str = "",
+        pci_slot: str = "",
+    ) -> None:
+        self._has_secondary = lower != "" or pci_slot == ""
+        self._upper = upper
+        self._lower = lower
+        self._pci_slot = pci_slot
+        self._ip_addr = ""
+        self._mac_addr = ""
+
+    def has_secondary(self) -> bool:
```

Comment:
Does "secondary" mean "lower" here? Make terms consistence. If we takes the upper and lower parts, make sure it's generic, not Azure only.

Reply:
And it can be a property and calculate from the condition every time. It can reduce the logic to maintain things consistent.

Reply:
yep, I'll change it